File: resources/expected/ros2/building_control_gen_mixed_inverted_topics/lax/src/building_control_cpp_pkg/src/base_code/tcp_fan_base_src.cpp
#include "building_control_cpp_pkg/base_headers/tcp_fan_base_src.hpp"

//========================================================
// Re-running Codegen will overwrite changes to this file
//========================================================

tcp_fan_base::tcp_fan_base() : Node("tcp_fan")
{
    cb_group_ = this->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
    subscription_options_.callback_group = cb_group_;

    // Setting up connections
    tcp_fan_fanCmd_subscription_ = this->create_subscription<building_control_cpp_pkg_interfaces::msg::FanCmd>(
        "tcp_tempControl_fanCmd",
        1,
        std::bind(&tcp_fan_base::handle_fanCmd, this, std::placeholders::_1), subscription_options_);

    tcp_fan_fanAck_publisher_ = this->create_publisher<building_control_cpp_pkg_interfaces::msg::FanAck>(
        "tcp_fan_fanAck",
        1);

}

//=================================================
//  C o m m u n i c a t i o n
//=================================================

void tcp_fan_base::put_fanAck(building_control_cpp_pkg_interfaces::msg::FanAck msg)
{
    tcp_fan_fanAck_publisher_->publish(msg);
}

File: resources/expected/ros2/python-building_control_gen_mixed/lax/src/building_control_py_pkg/building_control_py_pkg/base_code/tcp_fan_base_src.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from queue import Queue
from building_control_py_pkg.user_code.tcp_fan_src import *
from rclpy.callback_groups import ReentrantCallbackGroup
from building_control_py_pkg_interfaces.msg import FanCmd
from building_control_py_pkg_interfaces.msg import FanAck

#========================================================
# Re-running Codegen will overwrite changes to this file
#========================================================

class tcp_fan_base(Node):
    def __init__(self):
        super().__init__("tcp_fan")

        self.cb_group_ = ReentrantCallbackGroup()

        # Setting up connections
        self.tcp_fan_fanCmd_subscription_ = self.create_subscription(
            FanCmd,
            "tcp_fan_fanCmd",
            self.handle_fanCmd,
            1,
            callback_group=self.cb_group_)

        self.tcp_fan_fanAck_publisher_ = self.create_publisher(
            FanAck,
            "tcp_tempControl_fanAck",
            1)

    #=================================================
    #  C o m m u n i c a t i o n
    #=================================================

    def put_fanAck(self, msg):
        typedMsg = FanAck()
        typedMsg.data = msg
        self.tcp_fan_fanAck_publisher_.publish(typedMsg)


    #=================================================
    #  C o m p u t e    E n t r y    P o i n t
    #=================================================
    def handle_fanCmd(self, msg):
        raise NotImplementedError("Subclasses must implement this method")

